src/axom/core/tests/core_array.hpp

```diff
@@ -2359,3 +2359,103 @@ TEST(core_array, reserve_nontrivial_reloc_um)
   }
 }
 #endif
+
+// Regression test for a memory leak in axom::Array's copy/move/initialization
+template <typename T1, typename T2>
+auto make_arr_data(int SZ1, int SZ2)
+  -> std::pair<axom::Array<axom::Array<T1>>, axom::Array<axom::Array<T2>>>
+{
+  using Arr1 = axom::Array<axom::Array<T1>>;
+  using Arr2 = axom::Array<axom::Array<T2>>;
+
+  T1 val1 {};
+  T2 val2 {};
+
+  Arr1 arr1(SZ1, 2 * SZ1);
+  for(int i = 0; i < SZ1; ++i)
+  {
+    arr1[i].resize(i + 10);
+    arr1[i].shrink();
+    arr1[i].push_back(val1);
+  }
+
+  Arr2 arr2(SZ2, 2 * SZ2);
+  for(int i = 0; i < SZ2; ++i)
+  {
+    arr2[i].resize(i + 10);
+    arr2[i].shrink();
+    arr2[i].push_back(val2);
+  }
+  arr2.shrink();
+
+  return {std::move(arr1), std::move(arr2)};
+}
```

Comment:
Use-case modeled after a memory leak reported by LSAN on this function:
https://github.com/LLNL/Tribol/blob/5808ecc73d72a144c6e4d29d528e410fffe9ba81/src/redecomp/transfer/TransferByNodes.cpp#L227-L277